### PR TITLE
MueLu: Add more unittests for MueLu Utilities_kokkos

### PR DESCRIPTION
**Pull request by JacobDomagala**
Related to https://github.com/trilinos/Trilinos/issues/9743

### CODE REVIEW

**Comment by JacobDomagala** [comment]
There is still an issue with few tests when using `Scalar = std::complex<double>` and `Node = CudaSpace`

**Comment by JacobDomagala** [comment]
> There is still an issue with few tests when using `Scalar = std::complex<double>` and `Node = CudaSpace`

@jhux2  Ok seems that using nvcc 11.5 was the issue, as the PR testers passed. Feel free to review this PR.

File: packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp [hunk 1]
```diff
@@ -249,11 +249,11 @@ namespace MueLu {
 
         @return boolean array.  The ith entry is true iff row i is a Dirichlet row.
     */
-    static Kokkos::View<bool*, typename NO::device_type> DetectDirichletRows(const Matrix& A, const Magnitude& tol = Teuchos::ScalarTraits<SC>::zero(), const bool count_twos_as_dirichlet=false);
```
Comment (cgcgcg) [inline packages/muelu/src/Utils/MueLu_Utilities_kokkos_decl.hpp:252]: Maybe prefer
`Teuchos::ScalarTraits<typename Teuchos::ScalarTraits<SC>::magnitudeType>::zero()`?

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 2]
```diff
@@ -77,25 +100,551 @@ namespace MueLuTests {
...
     // CM Test
-    RCP<Xpetra::Vector<LocalOrdinal,LocalOrdinal,GlobalOrdinal,Node> > ordering
-      = MueLu::Utilities_kokkos<SC,LO,GO,NO>::CuthillMcKee(*A);
+    RCP<Xpetra::Vector<LocalOrdinal, LocalOrdinal, GlobalOrdinal, Node>> ordering = MueLu::Utilities_kokkos<SC, LO, GO, NO>::CuthillMcKee(*A);
 
+    RCP<Xpetra::Vector<LocalOrdinal, LocalOrdinal, GlobalOrdinal, Node>> ordering2 = MueLu::Utilities_kokkos<SC, LO, GO, NO>::ReverseCuthillMcKee(*A);
 
-    RCP<Xpetra::Vector<LocalOrdinal,LocalOrdinal,GlobalOrdinal,Node> > ordering2
-      = MueLu::Utilities_kokkos<SC,LO,GO,NO>::ReverseCuthillMcKee(*A);
+    TEST_EQUALITY(1, 1);
```
Comment (cgcgcg) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:110]: ?

**Review by cgcgcg — CHANGES_REQUESTED** [review]

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp (hunk 2 above)
Comment (JacobDomagala) [reply packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:110]: This is actually a leftover from original file (only formatting has changed for CuthillMcKee test). I should probably change that, totally forgot about this test

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 3]
```diff
@@ -55,47 +55,607 @@
 #include "MueLu_Utilities_kokkos.hpp"
 #include "MueLu_Version.hpp"
 
-namespace MueLuTests {
+namespace MueLuTests
+{
 
+  template <typename SC, typename LO, typename GO, typename NO>
+  void CreateDirchletRow(Teuchos::RCP<Xpetra::Matrix<SC, LO, GO, NO>> &A, LO localRowToZero, SC value = Teuchos::ScalarTraits<SC>::zero())
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:62]: Small typo, "CreateDirichletRow"

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 4]
```diff
@@ -55,47 +55,607 @@
...
+    CreateDirchletRow<Scalar, LocalOrdinal, GlobalOrdinal, Node>(A, localRowToZero);
+
+    auto drows = Utils_Kokkos::DetectDirichletRows(*A);
+    auto drowsHost = Kokkos::create_mirror_view(drows);
+    Kokkos::deep_copy(drowsHost, drows);
+
+    TEST_EQUALITY(drowsHost(localRowToZero), true);
+    TEST_EQUALITY(drowsHost(localRowToZero - 1), false);
+
+    CreateDirchletRow<Scalar, LocalOrdinal, GlobalOrdinal, Node>(A, localRowToZero, Teuchos::as<Scalar>(0.25));
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:146]: Small typo, "CreateDirichletRow"

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 5]
```diff
@@ -55,47 +55,607 @@
...
+    MUELU_TESTING_LIMIT_SCOPE(Scalar, GlobalOrdinal, Node);
+
+    using TST = Teuchos::ScalarTraits<Scalar>;
+    using Utils_Kokkos = MueLu::Utilities_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+    using MueLu_TestHelper_Factory = MueLuTests::TestHelpers_kokkos::TestFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+
+    auto A = MueLu_TestHelper_Factory::Build1DPoisson(100);
+
+    LocalOrdinal localRowToZero = 5;
+    CreateDirchletRow<Scalar, LocalOrdinal, GlobalOrdinal, Node>(A, localRowToZero);
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:178]: Small typo, "CreateDirichletRow"

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 6]
```diff
@@ -55,47 +55,607 @@
...
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:316]: Small typo, "CreateDirichletRow"

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 7]
```diff
@@ -55,47 +55,607 @@
...
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:348]: Small typo, "CreateDirichletRow"

File: packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp [hunk 8]
```diff
@@ -55,47 +55,607 @@
...
-  TEUCHOS_UNIT_TEST_TEMPLATE_4_INSTANT(Utilities_kokkos, CuthillMcKee, SC, LO, GO, NO)
+    using TST = Teuchos::ScalarTraits<Scalar>;
+    using Utils_Kokkos = MueLu::Utilities_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+    using MueLu_TestHelper_Factory = MueLuTests::TestHelpers_kokkos::TestFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
 
-#include <MueLu_ETI_4arg.hpp>
+    auto A = MueLu_TestHelper_Factory::Build1DPoisson(100);
+    LocalOrdinal localRowToZero = 5;
+
+    CreateDirchletRow<Scalar, LocalOrdinal, GlobalOrdinal, Node>(A, localRowToZero);
```
Comment (jhux2) [inline packages/muelu/test/unit_tests_kokkos/Utilities_kokkos.cpp:137]: Small typo, "CreateDirichletRow"

**Review by jhux2 — COMMENTED** [review]
Lgtm, just a couple typos

**Comment by JacobDomagala** [comment]
Thanks for reviews @cgcgcg @jhux2 !
I've applied the changes, let me know what you think.

**Review by jhux2 — APPROVED** [review]
lgtm, thanks

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**